api/middleware/contentLengthLimiterMiddleware.go

```diff
@@ -0,0 +1,59 @@
+package middleware
+
+import (
+	"errors"
+	"fmt"
+	"github.com/gin-gonic/gin"
+	"github.com/multiversx/mx-chain-go/api/shared"
+	"net/http"
+)
+
+type contentLengthLimiterMiddleware struct {
+	MaxSizeBytes int64
+}
+
+// NewContentLengthLimiterMiddleware will abort all requests that have Content-Length size bigger
+// than the one specified in config.
+func NewContentLengthLimiterMiddleware(maxSizeBytes int64) *contentLengthLimiterMiddleware {
+	return &contentLengthLimiterMiddleware{MaxSizeBytes: maxSizeBytes}
```

Comment:
should be checked against a min value in order to avoid config issuess, then return error.. maxSizeBytes > 1